tests/python_package_test/test_sklearn.py

```diff
@@ -1149,3 +1127,24 @@ def test_check_is_fitted(self):
         rnk.fit(X, y, group=np.ones(X.shape[0]))
         for model in models:
             check_is_fitted(model)
+
+
+def _tested_estimators():
+    for Estimator in [lgb.sklearn.LGBMClassifier, lgb.sklearn.LGBMRegressor]:
+        yield Estimator()
+
+
+@pytest.mark.skipif(
+    sk_version < "0.23.0", reason="scikit-learn version is less than 0.23"
+)
+@parametrize_with_checks(list(_tested_estimators()))
```

Comment:
Only the `parametrize_with_checks` was using `_xfail` in 0.23. It has the advantage to show all the test from `_yield` as with `parametrize` of `pytest`.

Reply:
However, it needs to be a pure function not embedded within a class